sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/batch/TransactionalBatch.java

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.batch;
+
+import com.azure.cosmos.implementation.RequestOptions;
```

Comment:
This class is in the public surface area but exposes implementation types (RequestOptions) in its surface area. Needs some refactoring to separate model/public surface area and implementation

Reply:
RequestOption is used in CosmosAsyncContainer. We need request options from user. I will create a separate one like TransactionalBatchRequestOptions like we have in C# and I can see similar model for CosmosItemRequestOption.

Reply:
Have fixed it. Please check if it seems fine.